jetty-server/src/main/java/org/eclipse/jetty/server/session/Session.java

```diff
@@ -481,6 +481,10 @@ public String getVHost()
     @Override
     public long getLastAccessedTime()
     {
+        if (isInvalid())
+        {
+            throw new IllegalStateException("Session not valid");
+        }
         try (Lock lock = _lock.lock())
```

Comment:
I think you should be doing the isInvalid() test inside the lock at line 488, otherwise the session could become invalid between line 484 and 486.

Reply:
done